### PR TITLE
Implement logging for albs-sign-file

### PR DESCRIPTION
**Pull request by Kwaizer**
Resolves: https://cloudlinux.atlassian.net/browse/AL-5608
- Added file hasher and SysLog class

### CODE REVIEW

File: sign/pgp/pgp.py [hunk 1]
```diff
@@ -68,6 +78,18 @@ async def sign(
                 timeout=1200,
                 withexitstatus=1,
             )
+            hash_after = hash_file(
+                    fd.name,
```
Comment (Korulag) [inline sign/pgp/pgp.py:82]: Incorrect indents

File: sign/log.py [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+    def sign_log(
+        self,
+        file_name: str,
+        hash_before: str,
+        hash_after: str,
+        pgp_keyid: str,
+    ):
+        self._logger.info(
+            ' Filename: %s. Hash before: %s. '
```
Comment (Korulag) [inline sign/log.py:24]: Remove whitespace in the beginning of the string

File: sign/utils/hashing.py [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+    ----------
+    checksum_type : str
+        Checksum type (e.g. sha1, sha256).
+
+    Returns
+    -------
+    _hashlib.HASH
+        Hashlib hashing function.
+    """
+    return hashlib.new('sha1' if checksum_type == 'sha' else checksum_type)
```
Comment (Korulag) [inline sign/utils/hashing.py:59]: Change default hasher to `sha256` since sha1 is too weak.

**Review by Korulag — CHANGES_REQUESTED** [review]

File: sign/pgp/pgp.py [hunk 4]
```diff
@@ -25,6 +29,7 @@ def __init__(self,
         self.max_upload_bytes = max_upload_bytes
         self.tmp_dir = tmp_dir
         self.__pass_db.ask_for_passwords()
+        self.__syslog = SysLog(tag_name='albs-sign-service')
```
Comment (soksanichenko) [inline sign/pgp/pgp.py:32]: Make a tag name configurable, because we use that service for CLBS also

**Review by soksanichenko — CHANGES_REQUESTED** [review]

File: sign/utils/hashing.py [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+    if isinstance(file_path, str):
+        with open(file_path, "rb") as fd:
+            feed_hasher(fd)
+    else:
+        file_path.seek(0)
+        feed_hasher(file_path)
+    return hasher.hexdigest()
+
+
+def get_hasher(checksum_type):
```
Comment (anfimovdm) [inline sign/utils/hashing.py:44]: unused parameter

File: sign/utils/hashing.py [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+    buff_size : int
+        Number of bytes to read at once.
+
+    Returns
+    -------
+    str
+        Checksum (hexadecimal digest) of the file.
+    """
+    if hasher is None:
+        hasher = get_hasher(hash_type)
```
Comment (anfimovdm) [inline sign/utils/hashing.py:25]: same as below

File: sign/utils/hashing.py [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+    Returns checksum (hexadecimal digest) of the file.
+
+    Parameters
+    ----------
+    file_path : str or file-like
+        File to hash. It could be either a path or a file descriptor.
+    hasher : hashlib.Hasher
+        Any hash algorithm from hashlib.
+    hash_type : str
+        Hash type (e.g. sha1, sha256).
```
Comment (anfimovdm) [inline sign/utils/hashing.py:15]: please remove it since we use only sha256 algorithm

**Review by anfimovdm — CHANGES_REQUESTED** [review]

**Review by anfimovdm — APPROVED** [review]

**Review by Korulag — APPROVED** [review]

**Review by soksanichenko — APPROVED** [review]

**Pull request merged**